Use `wp_body_open` for output of skip link

andersnoren opened the pull request:
Moves the output of the skip link from `header.php` to the `wp_body_open` action, matching the approach that will be used [in Twenty Eleven](https://core.trac.wordpress.org/ticket/47891). This also makes sure that `wp_body_open()` is triggered right after the `<body>`.

andersnoren commented:
Fixed #32, #16

joyously commented on functions.php:
```diff
@@ -328,6 +328,18 @@ function twentytwenty_get_custom_logo( $logo_theme_mod = 'custom_logo' ) {
+	function twentytwenty_skip_link() {
+		echo '<a class="skip-link faux-button" href="#site-content">' . esc_html__( 'Skip to the content', 'twentytwenty' ) . '</a>';
+	}
+	add_action( 'wp_body_open', 'twentytwenty_skip_link', 5 );
```
Shouldn't the priority be something larger than the default of 10?
The whole point of the new hook was for plugins that had to be first.

senadir replied:
do you mean that plugins need to render before this link or after it?

joyously replied:
I think before it.

aeonwp replied:
I thought the idea was to keep the skip link before whatever plugins put there.

carolinan replied:
I thought the idea was that the skip link needs to be before whatever plugins output there.

joyously replied:
Did you read the ticket for that new hook? The plugins needed it so they could be first.

joyously replied:
This confusion sounds like a good reason to leave the link in the template file and not use the hook at all.

pattonwebz commented:
Skip link should be before any other rendered content (so priority lower than default to prevent hook ordering issues). Content that does not render but wants to be first (aka Google tag manager) should use an even lower priority.

pattonwebz commented:
The PR looks correct to me in terms of how I imagine the prioritisation to work.

carolinan commented:
The link just needs to include the screen-reader-text class,then it can be merged.

carolinan commented:
Actually I believe we are also missing the fallback for wp_body_open?

andersnoren commented:
Oh right. I've updated the PR by removing the `function_exists()` conditional in header.php, and added the `wp_body_open()` shim suggested in https://make.wordpress.org/themes/2019/03/29/addition-of-new-wp_body_open-hook/.

pattonwebz commented:
This looks good to me, fallback working fine on pre 5.2 :)
